lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -833,6 +882,46 @@ def objects_for_user_or_attribute_and_operation(user_or_attribute, operation, op
       build_accessible_object_scope(filtered_associations, options)
     end
 
+    def objects_for_user_or_attribute_and_operations(user_or_attribute, operations, options)
+      return {} if operations.empty?
+      associations = associations_for_user_or_attribute(user_or_attribute, options)
+
+      # the operation set IDs and the object attribute IDs they connect to
+      # from the user_or_attribute's associations
+      opset_ids_to_objattr_ids = associations.pluck(
+        :operation_set_id,
+        :object_attribute_id
+      ).reduce(Hash.new { |h, k| h[k] = [] }) do |acc, (opset_id, objattr_id)|
```

Comment:
yea, i am kinda an `each_with_object` hater.. altho, i guess since it is now part of Ruby itself rather than just a Rails hack perhaps i can put away my hate.